File: capture-service/types/packet.go
package types

type PacketData struct {
	Timestamp   string `json:"timestamp"`
	SrcIP       string `json:"src_ip"`
	DstIP       string `json:"dst_ip"`
	Protocol    string `json:"protocol"`
	SrcPort     uint16 `json:"src_port"`
	DstPort     uint16 `json:"dst_port"`
	PacketSize  int    `json:"packet_size"`
	PacketType  string `json:"packet_type"`
	PayloadSize int    `json:"payload_size"`
}

File: analysis-service/src/types/packet.ts
export interface PacketData {
    source: string;
    destination: string;
    protocol: string;
    size: number;
    type: string;
    timestamp: string;
    src_ip: string;
    dst_ip: string;
    src_port: number;
    dst_port: number;
    packet_size: number;
    packet_type: string;
    payload_size: number;
}

export interface PacketAnalysisResult {
    packet: PacketData;
    score: number;
    causes: Array<{
        reason: string;
        score: number;
    }>;
    timestamp: string;
} 